frontend/hub/execution-environments/ExecutionEnvironmentForm.tsx

```diff
@@ -0,0 +1,333 @@
+              isDisabled
+            />
+          )}
+
+          {isRemote && (
+            <>
+              <PageFormTextInput<ExecutionEnvironmentFormProps>
+                name="upstream_name"
+                label={t('Upstream name')}
+                placeholder={t('Enter a upstream name')}
+                isRequired
+              />
+
+              <PageFormAsyncSelect<ExecutionEnvironmentFormProps>
+                name="registry"
+                label={t('Registry')}
+                placeholder={t('Select registry')}
+                query={query}
+                loadingPlaceholder={t('Loading registry...')}
+                loadingErrorText={t('Error loading registry')}
+                limit={page_size}
+                valueToString={(value: ExecutionEnvironment) => value.name}
+                openSelectDialog={registrySelector}
+                isRequired
+              />
+
+              <TagsSelector tags={tagsToInclude} setTags={setTagsToInclude} mode={'include'} />
+              <TagsSelector tags={tagsToExclude} setTags={setTagsToExclude} mode={'exclude'} />
+            </>
+          )}
+
+          <PageFormTextArea<ExecutionEnvironmentFormProps>
+            name="description"
+            label={t('Description')}
+            placeholder={t('Enter a description')}
+            isDisabled={mode == 'add'}
+          />
+        </HubPageForm>
+      )}
+    </PageLayout>
+  );
+}
+
+function validateName(name: string, t: TFunction<'translation', undefined>) {
+  const regex = /^([0-9A-Za-z._-]+\/)?[0-9A-Za-z._-]+$/;
+  if (regex.test(name)) {
+    return undefined;
+  } else {
+    return t(
+      `Container names can only contain alphanumeric characters, ".", "_", "-" and a up to one "/".`
+    );
+  }
+}
+
+type ExecutionEnvironmentFormProps = {
+  name: string;
+  upstream_name: string;
+  description?: string;
+  registry: Registry;
+  namespace?: { name: string };
+};
+
+type PayloadDataType = {
+  include_tags?: string[];
+  exclude_tags?: string[];
+  name: string;
+  upstream_name: string;
+  registry: string;
+};
+
+type Registry = {
+  id: string;
+  name: string;
+};
+
+function TagsSelector(props: {
+  tags: string[];
+  setTags: (tags: string[]) => void;
+  mode: 'exclude' | 'include';
+}) {
+  const [tagsText, setTagsText] = useState<string>('');
+  const { tags, setTags, mode } = props;
+  const { t } = useTranslation();
+
+  const label = mode == 'exclude' ? t('Add tag(s) to exclude') : t('Add tag(s) to include');
+  const label2 = mode == 'exclude' ? t('Currently excluded tags') : t('Currently included tags');
+
+  const chipGroupProps = () => {
+    const count = '${remaining}';
+    return {
+      collapsedText: t(`{{count}} more`, count.toString()),
+      expandedText: t(`Show Less`),
+    };
+  };
+
+  const addTags = () => {
+    if (tagsText == '' || !tagsText.trim().length) {
+      return;
+    }
+    const tagsArray = tagsText.split(',');
```

Comment:
```suggestion
    const tagsArray = tagsText.split(/\s+|\s*,\s*/).filter(Boolean);
```

split on any combination of commas and spaces, not just a singular comma (and throw away any empty fragments)

`"  foo    bar,baz,   quux  ,"` should resolve to `foo`, `bar`, `baz` and `quux` (instead of `  foo    bar`, `baz`, `  quux  ` and empty string)